src/_guides/language/effective-dart/usage.md

```diff
@@ -19,21 +19,58 @@ but *maintainers* of it sure will.
 
 These guidelines help you compose your program out of multiple files in a
 consistent, maintainable way. To keep these guidelines brief, they use "import"
-to cover `import` and `export` directives. The guidelines apply equally to both.
+to cover `import` and `export` directives. The guidelines apply equally to both. 
+
+### DON'T use library directives unless attaching doc comments or annotations
+
+[Library-level doc comments][] and [annotations][]
+must prepend a `library` declaration at the start of a file.
+There is no other reason to use the library directive. 
+
+{:.bad}
+<?code-excerpt "usage_bad.dart (library-dir)"?>
+{% prettify dart tag=pre+code %}
+
+library;
+{% endprettify %}
+
+{:.good}
+<?code-excerpt "docs_good.dart (library-doc)"?>
+{% prettify dart tag=pre+code %}
+/// A really great test library.
+@TestOn('browser')
+library;
+{% endprettify %}
+
+[Library-level doc comments]: /guides/language/effective-dart/documentation#consider-writing-a-library-level-doc-comment
+[annotations]: /guides/language/language-tour#metadata
+
+### DON'T explicitly name libraries
+
+Though appending a name to the `library` directive is technically possible,
+it is a legacy feature and discouraged. 
+
+A unique tag is generated for each library based on its path and filename.
+Naming libraries overrides their intrinsic URI, which can actually make it
+harder for tools to physically find the main library file you're referring to. 
+
+To indicate a specific library, the preferred, modern syntax
+is to use a URI string that points directly to the library file.
```

Comment:
I think you can remove this sentence because it's not clear *where* a user might be indicating and they may just get needlessly confused. The next guideline on `part of` does a good job of pointing on the place where this comes into play.